### PR TITLE
Some printf fixes

### PR DESCRIPTION
**Pull request by jonathangold**
govet picked up on some mistakes with printf (and errorf) so I went ahead and fixed them.

### CODE REVIEW

**Comment by purpleidea** [comment]
@jonathangold Is our `go vet` test case broken??

**Comment by jonathangold** [comment]
I think it's just the difference between go 1.9 and 1.10. I'll have a look at the test though.

File: engine/resources/virt.go [hunk 1]
```diff
@@ -403,11 +403,11 @@ func (obj *VirtRes) Watch() error {
 				}
 
 			} else {
-				return fmt.Errorf("unknown %s guest agent state: %v", obj, state)
+				return fmt.Errorf("unknown %s guest agent state: %v", obj.Name(), state)
```
[the remark this answers is not in the document]
Comment (jonathangold) [reply engine/resources/virt.go]: Maybe it should be %v instead of %s then?

[the remark this answers is not in the document]
Comment (jonathangold) [reply engine/resources/virt.go]: It's definitely getting picked up by go vet, not just our test https://goreportcard.com/report/github.com/purpleidea/mgmt#go_vet

[the remark this answers is not in the document]
Comment (jonathangold) [reply engine/resources/virt.go]: I may have only seen it on the goreportcard, and not locally. I can't reproduce the failure now anyway, so I dropped the commit from the pr.

We can always fix the false positive on goreportcard if it's a problem down the line.

**Comment by purpleidea** [comment]
Rebased to master (to include the travis fix) and merged! Thanks!

**Pull request closed without merging**